plugin/pkg/shoot/mutator/admission_test.go

```diff
@@ -652,5 +696,722 @@ var _ = Describe("mutator", func() {
 				})
 			})
 		})
+
+		DescribeTableSubtree("machine image", func(isCapabilityCloudProfile bool) {
+			var (
+				classificationPreview = gardencorev1beta1.ClassificationPreview
+
+				imageName1 = validMachineImageName
+				imageName2 = "other-image"
+
+				expiredVersion                                  = "1.1.1"
+				expiringVersion                                 = "1.2.1"
+				nonExpiredVersion                               = "2.0.0"
+				latestNonExpiredVersionThatSupportsCapabilities = "2.0.1"
+				latestNonExpiredVersion                         = "2.1.0"
+				previewVersion                                  = "3.0.0"
+
+				cloudProfileMachineImages []gardencorev1beta1.MachineImage
+			)
+
+			BeforeEach(func() {
+				machineCapabilities := []gardencorev1beta1.CapabilityDefinition{
+					{Name: "architecture", Values: []string{v1beta1constants.ArchitectureAMD64}},
+					{Name: "someCapability", Values: []string{"value1", "value2"}},
+				}
+				cloudProfileMachineImages = []gardencorev1beta1.MachineImage{
+					{
+						Name: validMachineImageName,
+						Versions: []gardencorev1beta1.MachineImageVersion{
+							{
+								ExpirableVersion: gardencorev1beta1.ExpirableVersion{
+									Version:        previewVersion,
+									Classification: &classificationPreview,
+								},
+								CapabilityFlavors: []gardencorev1beta1.MachineImageFlavor{
+									{Capabilities: gardencorev1beta1.Capabilities{"architecture": []string{v1beta1constants.ArchitectureAMD64}}},
+									{Capabilities: gardencorev1beta1.Capabilities{"architecture": []string{v1beta1constants.ArchitectureARM64}}},
+								},
+							},
+							{
+								ExpirableVersion: gardencorev1beta1.ExpirableVersion{
+									Version: latestNonExpiredVersion,
+								},
+								CapabilityFlavors: []gardencorev1beta1.MachineImageFlavor{
+									{Capabilities: gardencorev1beta1.Capabilities{
+										"architecture":   []string{v1beta1constants.ArchitectureAMD64},
+										"someCapability": []string{"value1"},
+									}},
+									{Capabilities: gardencorev1beta1.Capabilities{
+										"architecture":   []string{v1beta1constants.ArchitectureARM64},
+										"someCapability": []string{"value1"},
+									}},
+								}},
```

Comment:
```suggestion
								},
							},
```
The same is valid for some blocks below